x-pack/plugins/security_solution/public/app/home/global_header/index.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiHeaderSection,
+  EuiHeaderLinks,
+  EuiHeaderLink,
+  EuiHeaderSectionItem,
+} from '@elastic/eui';
+import React, { useEffect, useMemo } from 'react';
+import { createPortalNode, OutPortal, InPortal } from 'react-reverse-portal';
+import { i18n } from '@kbn/i18n';
+
+import { AppMountParameters } from '../../../../../../../src/core/public';
+import { toMountPoint } from '../../../../../../../src/plugins/kibana_react/public';
+import { MlPopover } from '../../../common/components/ml_popover/ml_popover';
+import { useKibana } from '../../../common/lib/kibana';
+import { ADD_DATA_PATH, APP_DETECTIONS_PATH } from '../../../../common/constants';
+
+const BUTTON_ADD_DATA = i18n.translate('xpack.securitySolution.globalHeader.buttonAddData', {
+  defaultMessage: 'Add data',
+});
+
+/**
+ * This component uses the reverse portal to add the Add Data and ML job settings buttons on the
+ * right hand side of the Kibana global header
+ */
+export const GlobalHeader = React.memo(
+  ({ setHeaderActionMenu }: { setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'] }) => {
+    const portalNode = useMemo(() => createPortalNode(), []);
+    const { http } = useKibana().services;
+
+    useEffect(() => {
+      let unmount = () => {};
+
+      setHeaderActionMenu((element) => {
+        const mount = toMountPoint(<OutPortal node={portalNode} />);
+        unmount = mount(element);
+        return unmount;
+      });
+
+      return () => {
+        portalNode.unmount();
+        unmount();
+      };
+    }, [portalNode, setHeaderActionMenu]);
+
+    const basePath = http.basePath.get();
+    return (
+      <InPortal node={portalNode}>
+        <EuiHeaderSection side="right">
+          {window.location.pathname.includes(APP_DETECTIONS_PATH) && (
+            <EuiHeaderSectionItem>
+              <MlPopover />
```

Comment:
Is it possible at all to change this from a popover to a flyout? It's so tall 😆 
<img width="1009" alt="Screen Shot 2021-06-15 at 16 40 30 PM" src="https://user-images.githubusercontent.com/549577/122120541-8dd40d00-cdf8-11eb-9020-5f79e50328d8.png">

Reply:
Lol, that's a conversation I'll need to have with the PM's. We do already have a side panel `flyout` component, but not sure if they'd prefer it there. Personally, I think a modal would be perfect here. 🤷🏾‍♂️

Reply:
I mean it can just be an `EuiFlyout` directly too in stead of a popover? I even think there's too much content for a modal.

Reply:
Thanks for highlighting this @cchaos! 🙂 This has been on our radar for a bit but hasn't been prioritized just yet. I've created https://github.com/elastic/kibana/issues/102837 to capture this effort and for us to find a better dedicated home for the Security Solution ML Job management UI.

cc @MikePaquette, @paulewing and @yiyangliu9286